app/uk/gov/hmrc/helptosavefrontend/connectors/NSIConnector.scala

```diff
@@ -70,44 +68,55 @@ class NSIConnectorImpl @Inject() (conf: Configuration, auditor: HTSAuditor, metr
       ()
     )
 
-    val timeContext: Timer.Context = timer.time()
+    val timeContext: Timer.Context = metrics.nsiAccountCreationTimer.time()
 
     httpProxy.post(nsiUrl, userInfo, Map(nsiAuthHeaderKey → nsiBasicAuth))
       .map[SubmissionResult] { response ⇒
         val time = timeContext.stop()
 
         response.status match {
           case Status.CREATED ⇒
-            auditor.sendEvent(new ApplicationSubmittedEvent(appName, userInfo))
+            auditor.sendEvent(ApplicationSubmittedEvent(appName, userInfo))
             logger.info(s"Received 201 from NSI, successfully created account for ${userInfo.nino} ${timeString(time)}")
             SubmissionSuccess()
 
-          case Status.BAD_REQUEST ⇒
-            logger.warn(s"Failed to create an account for ${userInfo.nino} due to bad request ${timeString(time)}")
-            handleBadRequestResponse(response)
-
-          case Status.INTERNAL_SERVER_ERROR ⇒
-            logger.warn(s"Received 500 from NSI, failed to create account for ${userInfo.nino} as there was an " +
-              s"internal server error ${timeString(time)}")
-            handleBadRequestResponse(response)
-
-          case Status.SERVICE_UNAVAILABLE ⇒
-            logger.warn(s"Received 503 from NSI, failed to create account for ${userInfo.nino} as NSI " +
-              s"service is unavailable ${timeString(time)}")
-            handleBadRequestResponse(response)
-
           case other ⇒
-            logger.warn(s"Unexpected error during creating account for ${userInfo.nino}, status : $other ${timeString(time)}")
-            SubmissionFailure(None, s"Something unexpected happened; response body: ${response.body}", other.toString)
+            handleErrorStatus(other, response, userInfo.nino, time)
         }
       }.recover {
         case e ⇒
           val time = timeContext.stop()
+          metrics.nsiAccountCreationErrorCounter.inc()
+
           logger.warn(s"Encountered error while trying to create account ${timeString(time)}", e)
           SubmissionFailure(None, "Encountered error while trying to create account", e.getMessage)
       }
   }
 
+  private def handleErrorStatus(status: Int, response: HttpResponse, nino: NINO, time: Long) = {
+    metrics.nsiAccountCreationErrorCounter.inc()
+
+    status match {
+      case Status.BAD_REQUEST ⇒
+        logger.warn(s"Failed to create an account for $nino due to bad request ${timeString(time)}")
+        handleBadRequestResponse(response)
+
+      case Status.INTERNAL_SERVER_ERROR ⇒
+        logger.warn(s"Received 500 from NSI, failed to create account for $nino as there was an " +
+          s"internal server error ${timeString(time)}")
+        handleBadRequestResponse(response)
+
+      case Status.SERVICE_UNAVAILABLE ⇒
+        logger.warn(s"Received 503 from NSI, failed to create account for $nino as NSI " +
+          s"service is unavailable ${timeString(time)}")
+        handleBadRequestResponse(response)
+
+      case other ⇒
```

Comment:
this is still an error, don't we want to increment counters here ?

Reply:
I moved the increment of the counter above of the pattern match so it will get incremented in this case